Documentation/ApiOverview/Backend/JavaScript/ES6/Index.rst

```diff
@@ -6,29 +6,111 @@
+a polyfill is available for Firefox and Safari and included by TYPO3 core
+and applied whenever an importmap is emitted.
+
+For security reasons importmap configuration is only emitted when the modules
+are actually used, that means when a module has been added to the current
+page response via via :php:`PageRenderer->loadJavaScriptModule()` or
+:php:`JavaScriptRenderer->addJavaScriptModuleInstruction()`.
+Exposing all module configurations is possible via
+:php:`JavaScriptRenderer->includeAllImports()`, but that should only be
+done in backend context for logged in users, to avoid disclosing installed
+extensions to anonymous visitors.
+
+Configuration
+=============
+
+A simple configuration example for an extension that maps
+the `Public/JavaScript` folder to an import prefix `@vendor/my-extensions`:
+
+..  literalinclude:: _JavaScriptModulesSimple.php
+    :caption: EXT:my_extension/Configuration/JavaScriptModules.php
 
-    *   :ref:`ext_core:feature-96510`
-    *   :ref:`ext_core:feature-98431-1664652179`
+Complex configuration example containing recursive-lookup exclusions,
+third-party library definitions and overwrites:
 
+..  literalinclude:: _JavaScriptModulesExtended.php
+    :caption: EXT:my_extension/Configuration/JavaScriptModules.php
 
 .. _backend-javascript-es6-loading:
 
 Loading ES6
 ===========
 
-..  note::
-    This chapter is still work in progress.
+A module can be added to the current page response either via
+:php:`PageRenderer` or as :php:`JavaScriptModuleInstruction` via
+:php:`JavaScriptRenderer`:
+
+..  literalinclude:: _PageRendererJavaScriptLoading.php
+    :caption: EXT:my_extension/Classes/SomeNamespace/SomeClass.php
+
+In Fluid template the `includeJavaScriptModules` property of the
+:html:`<f:be.pageRenderer>` ViewHelper may be used:
+
+..  literalinclude:: _BackendFluidTemplate.html
+    :caption: EXT:my_extension/Resources/Private/Backend/Templates/SomeTemplate.html
+
+Some tips on ES6
+================
+
+In the TYPO3 Core there are no ES6 JavaScript files created directly. JavaScript
+is created as TypeScript module which is then converted to ES6 JavaScript
+during the build process. However TypeScript and ES6 are quite similar, you
+can therefore look into those files for reference. The TypeScript files can be
+found on GitHub at
+`Build/Sources/TypeScript <https://github.com/TYPO3/typo3/tree/main/Build/Sources/TypeScript>`__.
+
+For an example on a ES6 JavaScript file have a look at the JavaScript example in
+the :ref:`LinkHandler Tutorial <tutorial_backend_link_handler_javascript>` or
+the example in the :ref:`Notification API <notification_api>`.
+
+For a real life example on how to introduce ES6 modules into a large extension
+see this commit for EXT:news: `[TASK] Add support for TYPO3 v12 ES6
+modules <https://github.com/bnf/news/commit/f8e196b67ceaa2f56699fbf464080dde668ad526>`__.
+
+Using JQuery
+------------
+
+There is a general tendency to remove dependencies to jQuery in the Core whenever
```

Comment:
Hmm, would rephrase that, something like: "In the TYPO3 Core usage of jQuery is eliminated step-by-step as the functionality is provided by native JavaScript nowadays."

Reply:
Ok, go ahead